cgd_tx_eff/src/edu/ohsu/compbio/txeff/util/refseq_to_ccds.py

```diff
@@ -0,0 +1,121 @@
+
+        stream_format = logging.Formatter(logging_format)
+        stream_handler.setFormatter(stream_format)
+        stream_handler.setLevel(logging.INFO)
+        logger.addHandler(stream_handler)
+        self.logger = logger
+        
+    def get_mappings_from_gff(self, gff_file_name: str):
+        '''
+        Read the GFF file and return all refseq-to-ccds mappings 
+        '''
+        refseq_ccds_map = defaultdict(set)
+        
+        limit_info = dict(gff_type=['CDS'])
+        with open(gff_file_name, 'r') as gff_file:
+            self.logger.info(f"Reading {gff_file_name}")
+            
+            for rec in GFF.parse(gff_file, limit_info=limit_info):
+                for feature in rec.features:
+                    if feature.id.startswith('rna-'):
+                        
+                        # Remove the 'rna-' prefix and the occasional "-n" suffix (eg rna-NM_123.2-1)
+                        ref_seq_id_parts = feature.id.split('-')                    
+                        ref_seq_id = ref_seq_id_parts[1]
+                        
+                        if feature.sub_features:
+                            for sub_feature in feature.sub_features:                                
+                                dbXrefs = sub_feature.qualifiers.get('Dbxref')
+                                for dbxref in dbXrefs:
+                                    if(dbxref.startswith('CCDS')):
+                                        ccds_id = dbxref.split(':')[1]
+                                        refseq_ccds_map[ref_seq_id].add(ccds_id)
+                                        self.logger.debug(f"{ref_seq_id} --> {ccds_id}")
+                                    
+        self.logger.info(f"Created RefSeq-CCDS map of size {len(refseq_ccds_map)}")
+        return refseq_ccds_map
+    
+    def get_mappings_from_csv(self, csv_file_name: str):
+        '''
+            Read the csv file created by write_mappings, and return thm in a map object.
+        '''
+        refseq_ccds_map = defaultdict()
+        
+        with open(csv_file_name) as csv_file:
+            reader = csv.DictReader(csv_file)
+            for row in reader:
+                if refseq_ccds_map.get(row['refseq_id']) is not None:
+                    raise Exception(f"RefSeq id {row['refseq_id']} is already mapped to {refseq_ccds_map.get(row['refseq_id'])}. Cannot add additional mapping to {row['ccds_id']}")
+                
+                refseq_ccds_map[row['refseq_id']] = row['ccds_id']
+        
+        return refseq_ccds_map
+        
+        
+    def write_mappings(self, refSeq_to_ccds: map, csv_file_name: str):
+        '''
+        Write the RefSeq-to-CCDS map to CSV file
+        '''
+        fields = ['refseq_id', 'ccds_id']
+        with open(csv_file_name, 'w') as csvfile:
+            csv_writer = csv.writer(csvfile)
+            csv_writer.writerow(fields)
+            for refseq_id in refSeq_to_ccds.keys():
```

Comment:
Don't need keys() here, just use "in".  Can len(refSeq_to_ccds.get(refseq_id)) > 1 ever be true?